.github/workflows/reusable_pre_release.yaml

```diff
@@ -0,0 +1,211 @@
+name: Pre-Release
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: "Git tag vX.X.X for the Release. E.g: 'v1.0.1'"
+        required: true
+        type: string
+      integration:
+        description: "Integration name (without the prefix 'nri-'). E.g: 'redis'"
+        required: true
+        type: string
+      test_package:
+        description: Defines if the package should be tested using the 'newrelic/integrations-pkg-test-action' action.
+        type: boolean
+        required: false
+        default: true
+      publish_schema:
+        description: Defines the schema for the publish_action.
+        type: string
+        required: false
+        default: "ohi"
+      publish_schema_url:
+        description: Defines the schema_url for the publish_action (it requires publish_schema='custom')
+        type: string
+        required: false
+        default: ""
+      run_integration_nix:
+        required: false
+        type: boolean
+        default: true
+      run_test_windows:
+        required: false
+        type: boolean
+        default: true
+      run_build-win-packages:
+        description: "Build Windows packages"
+        required: false
+        type: boolean
+        default: true
+      win_package_type:
+        description: "Package type (msi,exe). Integrations that wraps other binaries are packages in exe (nri-kafka)"
+        default: msi
+        required: false
+        type: string
+    secrets:
+      OHAI_AWS_ROLE_ARN_STAGING:
+        required: true
+      OHAI_AWS_ROLE_SESSION_NAME_STAGING:
+        required: true
+      OHAI_AWS_ACCESS_KEY_ID_STAGING:
+        required: true
+      OHAI_AWS_SECRET_ACCESS_KEY_STAGING:
+        required: true
+      OHAI_GPG_PASSPHRASE:
+        required: true
+      OHAI_GPG_PRIVATE_KEY_BASE64:
+        required: true
+      OHAI_PFX_CERTIFICATE_BASE64:
+        required: false
+      OHAI_PFX_PASSPHRASE:
+        required: false
+      COREINT_SLACK_TOKEN:
+        required: true
+      COREINT_SLACK_CHANNEL:
+        required: true
+
+env:
+  # Build scripts inputs are mixed between this env vars and config args.
+  REPO_FULL_NAME: ${{ github.event.repository.full_name }}
+  TAG: ${{ inputs.tag }}
+  INTEGRATION: ${{ inputs.integration }}
```

Comment:
are these needed?

Reply:
yes, as the comment says , these are sadly expected to be present as env vars by the build scripts

Reply:
Sorry for the noise Somehow I missed such comment